src/controllers/trip.controller.js

```diff
@@ -328,5 +331,28 @@ class TripController {
       return response.errorMessage(res, error.message, 500);
     }
   }
+
+  /** Function to return the number of trips created by a user and breaks them down into
+   * types
+   *
+   * @param {*} req the request sent to the server
+   * @param {*} res the response
+   * @returns {*} the data found and the message
+   */
+  static async getTripStatistics(req, res) {
+    try {
+      let sum = 0;
+      const { startDate } = req.query;
+      const { id, firstName } = req.user;
+      const details = await tripService.findTripsCreatedByuser(id, startDate);
+      details.map((detailedTrip) => {
```

Comment:
@jo-rdan The correct array function to use here is reduce(), instead of using map().